src/nvim/globals.h

```diff
@@ -503,8 +503,8 @@ EXTERN int stdout_isatty INIT(= true);
 EXTERN int stdin_fd INIT(= -1);
 
 // true when doing full-screen output, otherwise only writing some messages.
-// volatile because it is used in a signal handler.
-EXTERN volatile int full_screen INIT(= false);
+// no need volatile because callback not called directly from a signal handlers.
+EXTERN int full_screen INIT(= false);
```

Comment:
can skip the comment here. `deadly_signal` is the most important one because that's where contributors might wonder why we have non-trivial code in a signal handler.

```suggestion
EXTERN int full_screen INIT(= false);
```